util/collate/general_ci.go

```diff
@@ -0,0 +1,295 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collate
+
+type generalCICollator struct {
+}
+
+func sign(i int) int {
+	if i < 0 {
+		return -1
+	} else if i > 0 {
+		return 1
+	}
+	return 0
+}
+
+// Compare implement Collator interface.
+func (gc *generalCICollator) Compare(a, b string) int {
+	aRune := []rune(a)
```

Comment:
My concern is converting & compare the rune one by one would be faster than this, can you make a benchmark to make a comparison?